base/strings/basic.jl

```diff
@@ -62,6 +62,7 @@ convert(::Type{Array{UInt8}}, s::AbstractString) = String(s).data
 convert(::Type{String}, s::AbstractString) = String(s)
 convert(::Type{Vector{Char}}, s::AbstractString) = collect(s)
 convert(::Type{Symbol}, s::AbstractString) = Symbol(s)
+convert(::Type{String}, s::Symbol) = unsafe_wrap(String, Cstring(s))
```

Comment:
It's not like someone can't muck around with the `Symbol` contents anyway using `Cstring` directly, though. Mutating the `data` field of a string should be a no-go unless you make the string yourself.